google-http-client/src/test/java/com/google/api/client/http/HttpResponseTest.java

```diff
@@ -457,4 +459,40 @@ public LowLevelHttpResponse execute() throws IOException {
         "it should not decompress stream",
         request.execute().getContent() instanceof GZIPInputStream);
   }
+
+  public void testGetContent_gzipEncoding_finishReading() throws IOException {
+    byte[] dataToCompress = "abcd".getBytes(StandardCharsets.UTF_8);
+    byte[] mockBytes;
+    try (ByteArrayOutputStream byteStream = new ByteArrayOutputStream(dataToCompress.length)) {
+      GZIPOutputStream zipStream = new GZIPOutputStream((byteStream));
+      zipStream.write(dataToCompress);
+      zipStream.close();
```

Comment:
We should probably also have a case where there's extra data on the stream after the gzipped content, since handling that is the purpose of this PR.